File: spondereServer/entity/academicClass.py
from typing import Optional
from pydantic.main import BaseModel

class AcademicClass(BaseModel):
    id:int = None
    groupID:int = None
    titleClass:str = None
    descriptionClass:str = None
    beginDate: Optional[str] = None
    endDate: Optional[str] = None
    longitude: Optional[float] = None
    latitude: Optional[float] = None
    activeValidation:bool= None
    validationByQrCode:bool = None
    validationByBLE:bool = None
    blockedAttendance:bool = None
    validationCode:Optional[str] = None

    class Config:
        schema_extra = {
            "example":{
                "groupID": 1,
                "titleClass": "Integral",
                "descriptionClass": "Aula 2.",
                "beginDate": "2022-10-19 10:10:00+2",
                "endDate": "2022-10-19 12:00:00+2",
                "activeValidation": False,
                "validationByQrCode": False,
                "validationByBLE": False,
                "blockedAttendance": False
            }
        }

File: protótipo/Spondere/app/src/main/java/com/spondere/entity/AcademicClass.kt
package com.spondere.entity

import com.fasterxml.jackson.annotation.JsonAlias
import com.fasterxml.jackson.annotation.JsonProperty
import java.io.Serializable

data class AcademicClass(
    @JsonProperty ("id")
    var id:Int,
    @JsonProperty ("groupid")
    @JsonAlias("groupID")
    val groupID:Int,
    @JsonProperty ("titleclass")
    @JsonAlias("titleClass")
    val titleClass:String,
    @JsonProperty ("descriptionclass")
    @JsonAlias("descriptionClass")
    val descriptionClass:String?,
    @JsonProperty ("begindate")
    @JsonAlias("beginDate")
    val beginDate: String?,
    @JsonProperty ("enddate")
    @JsonAlias("endDate")
    val endDate:String?,
    @JsonProperty ("longitude")
    val longitude:Double?,
    @JsonProperty ("latitude")
    val latitude:Double?,
    @JsonProperty ("activevalidation")
    @JsonAlias("activeValidation")
    val activeValidation:Boolean,
    @JsonProperty ("validationbyqrcode")
    @JsonAlias("validationByQrCode")
    val validationByQrCode:Boolean,
    @JsonProperty ("validationbyble")
    @JsonAlias("validationByBLE")
    val validationByBLE:Boolean,
    @JsonProperty ("blockedattendance")
    @JsonAlias("blockedAttendance")
    var blockedAttendance:Boolean,
    @JsonProperty ("validationcode")
    @JsonAlias("validationCode")
    val validationCode:String?):Serializable
